rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/moderationrequest/Sw360ModerationRequestService.java

```diff
@@ -447,4 +455,34 @@ public Integer getOpenCriticalCrCountByGroup(String group) {
             return 0;
         }
     }
+
+    public RequestStatus deleteModerationRequestInfo(@NotNull User sw360User, @NotNull String id,
+			@NotNull ModerationRequest moderationRequest, ModerationState moderationState)
+            throws TTransportException, TException {
+        RequestStatus requestStatus = null;
+        Set<String> moderators = moderationRequest.getModerators();
+        String requestingUser = moderationRequest.getRequestingUser();
+        if (moderators.contains(sw360User.getEmail())) {
+            if (moderationRequest.getModerators().contains(sw360User.getEmail())) {
+                if (moderationRequest.getModerationState() == ModerationState.REJECTED
+                        || moderationRequest.getModerationState() == ModerationState.APPROVED) {
+                    requestStatus = getThriftModerationClient().deleteModerationRequest(id, sw360User);
+                }
+            } else if (!sw360User.getEmail().equals(requestingUser)) {
+                if ((moderationRequest.getModerationState() == ModerationState.PENDING
+                        || moderationRequest.getModerationState() == ModerationState.INPROGRESS)) {
+                    requestStatus = requestStatus.FAILURE;
+                }
+            } else {
+                requestStatus = null;
+            }
```

Comment:
If the `sw360User` is neither moderator, nor the requesting user, shouldn't the `requestStatus` be `FAILURE`? Why is it `null`?

Reply:
put this condition because if the user adds moderators dynamically (e.g., modifies the moderators set or its source), we need to ensure the check accurately reflects the current state.